addons/purchase/models/account_invoice.py

```diff
@@ -130,7 +130,9 @@ class AccountMoveLine(models.Model):
     """ Override AccountInvoice_line to add the link to the purchase order line it is related to"""
     _inherit = 'account.move.line'
 
+    move_type = fields.Selection(related='move_id.move_type', readonly=True, store=True)
```

Comment:
If you store it, you'll need a migration script ;)

Also, can't you just write move_id.move_type in the domains directly, avoiding creating this field?